File: leetcode/two_pointers/dart/remove_duplicates_from_sorted_array_ii.dart
class Solution {
  int removeDuplicates(List<int> nums) {
    if (nums.length <= 2) {
      return nums.length;
    }
    int left = 0;
    for (var element in nums) {
      if (left < 2 || nums[left - 2] == element) {
        nums[left++] = element;
      }
    }
    return left;
  }
}

void main(List<String> args) {
  final nums = [0, 0, 1, 1, 2, 3, 3];
  final object = Solution();
  print(object.removeDuplicates(nums));
}

File: leetcode/two_pointers/go/remove_duplicates_from_sorted_array_ii.go
package two_pointers

func removeDuplicates(nums []int) int {
	left := 0
	for _, v := range nums {
		if left < 2 || nums[left - 2] == v {
			nums[left] = v
			left++
		}
	}
	return left
}

func main() {
	nums := []int{0,0,1,1,2,3,3}
	fmt.Println(removeDuplicates(nums))

}